src/addons/dragAndDrop/DropWrappers.js

```diff
@@ -0,0 +1,173 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { DropTarget } from 'react-dnd'
+import cn from 'classnames'
+import noop from 'lodash/noop'
+
+import { accessor } from '../../utils/propTypes'
+import { accessor as get } from '../../utils/accessors'
+import dates from '../../utils/dates'
+import BigCalendar from '../../index'
+
+function getEventDropProps(start, end, dropDate, type) {
+  // Calculate duration between original start and end dates
+  const duration = dates.diff(start, end)
+
+  /*
+   * If the event is dropped in a "Day" cell, preserve an event's start time by extracting the hours and minutes off
+   * the original start date and add it to newDate.value
+   * 
+   * note: this behavior remains for backward compatibility, but might be counter-intuitive to some:
+   * dragging an event from the grid to the day header might more commonly mean "make this an allDay event
+   * on that day" - but the behavior here implements "keep the times of the event, but move it to the
+   * new day".
+   * 
+   * To permit either interpretation, we embellish a new `allDay` parameter which determines whether the
+   * event was dropped on the day header or not.
+   */
+
+  const allDay = type === 'dateCellWrapper'
+  const nextStart = allDay ? dates.merge(dropDate, start) : dropDate
+  const nextEnd = dates.add(nextStart, duration, 'milliseconds')
+
+  return {
+    allDay,
+    start: nextStart,
+    end: nextEnd,
+  }
+}
+
+class DropWrapper extends React.Component {
+  static propTypes = {
+    connectDropTarget: PropTypes.func.isRequired,
+    type: PropTypes.string,
+    isOver: PropTypes.bool,
+  }
+
+  static contextTypes = {
+    onEventDrop: PropTypes.func,
+    onEventResize: PropTypes.func,
+    dragDropManager: PropTypes.object,
+    startAccessor: accessor,
+    endAccessor: accessor,
+    step: PropTypes.number,
+  }
+
+  // constructor(...args) {
```

Comment:
This commented out stuff has been here for a while and I am wondering if it's still needed. Thoughts @jquense?

Reply:
sorry about that, its my own personal reminder for how to structure the react-dnd stuff so that the drop areas align with the shap of the even vs just where the mouse is over (which i think is more intuitive feeling) I left it there to save my place, but  haven't had the time to explore it any further. We can remove it (git has is saved obviously) unless someone wants to pick it up ;)

Reply:
I sure would love to see the drop target be based on the mouse-relative location, not the top of the event. I agree that the current behavior isn't the best user experience. I ran out of energy (and competence :-)) on this so didn't tackle it.